### ISSUE TITLE
semanticodec unquantized concat order

### ISSUE THREAD

**Issue by sin168** [wuzhiyue111/Codec-Evaluation#26]
def extract_feature(self, waveforms, expect_lenth: torch.Tensor = None):
        """
            extract features from codec
            waveforms: [B, T]
            return: [B*n_segments, D, T]
        """
        length = torch.ones(waveforms.shape[0])
        all_features = self.codec(waveforms, length)

        if self.codec_name == 'semanticodec':
            assert expect_lenth is not None, "expect_lenth is required for semanticodec"
            if all_features.dim() == 4:
                all_features = rearrange(all_features, 'b d c t -> b (d c) t')
            max_length = expect_lenth
            all_features = all_features[:, :, :int(max_length)]

        return all_features
